yaksh/templates/exam.html

```diff
@@ -1,86 +1,147 @@
                 {% endif %}
                 {% endif %}
                 {% if qid in paper.get_questions_answered %}
-                <li><a style="background-color:#B4B8BA; width:25%" href="#" data-toggle="tooltip"
+                <a class="  btn btn-outline-success " style="background-color:#B4B8BA;" href="#" data-toggle="tooltip"
                 onclick="call_skip('{{ URL_ROOT }}/exam/{{ question.id }}/skip/{{ qid.id }}/{{ paper.attempt_number }}/{{ module.id }}/{{ paper.question_paper.id }}/{{course.id}}/')"
-                title="{{ qid.description }}">{{ forloop.counter }}</a></li>
+                title="{{ qid.description }}">{{ forloop.counter }}</a>
                 {% endif %}
                 {% else %}
                 {% if qid.id == question.id %}
-                <li class="active"><a style="width:25%" data-toggle="tooltip" title="{{ qid.description|striptags }}">{{ forloop.counter }}</a></li>
+                <a class="active btn btn-outline-success " data-toggle="tooltip" title="{{ qid.description|striptags }}">{{ forloop.counter }}</a>
                 {% elif qid in paper.get_questions_answered %}
-                <li><a style="background-color:#B4B8BA; width:25%" href="#" data-toggle="tooltip"
+                <a class=" btn btn-outline-success " style="background-color:#B4B8BA;" href="#" data-toggle="tooltip"
                 onclick="call_skip('{{ URL_ROOT }}/exam/{{ question.id }}/skip/{{ qid.id }}/{{ paper.attempt_number }}/{{ module.id }}/{{ paper.question_paper.id }}/{{course.id}}/')"
-                title="{{ qid.description }}">{{ forloop.counter }}</a></li>
+                title="{{ qid.description }}">{{ forloop.counter }}</a>
                 {% else %}
-                <li  class="disabled"><a style="width:25%" data-toggle="tooltip" title="{{ qid.description|striptags }}">{{ forloop.counter }}</a></li>
+                <a class="disabled btn btn-outline-success " data-toggle="tooltip" title="{{ qid.description|striptags }}">{{ forloop.counter }}</a>
                 {% endif %}
                 {% endif %}
                 {% endfor %}
-            </ul>
-            <p>Question(s) left: <b>{{ paper.questions_left }}</b></p>
-            <br><br><br>
-            {% block learning_units %}
-            {% endblock %}
-        </div>
-        <div class="col-sm-9 col-sm-offset-3 col-md-10 col-md-offset-2 main">
-                <div class="row">
-                {% block main %}
-                {% endblock %}
+                </div>
+                <br>
+          <p class="text-center">Question(s) left: <b>{{ paper.questions_left }}</b></p>
+          </div>
+          <br>
+          <div class="">
+            <div class="">
+              <div class="">
+                {% for types in question_types %}
+                {% get_questions_by_type  paper.get_all_ordered_questions types as questions_by_type %}  
+                  {% if types == "mcq" %}  Single Correct Choice 
```

Comment:
Personally, I like the idea of having a legend-like information table, being able to see which all questions belong to which question type. However the implementation here is slightly unpolished. May be this could be made better.

Reply:
Will raise as in issue for this.